extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalogAdapter.java

```diff
@@ -860,8 +835,456 @@ private Table readTableInternal(
+            if (addSnapshot) {
+                commit(icebergTable, newSpecAndSchema, appendFiles, overwrite && !newTableCreated, verifySchema);
+            }
+            return appendFiles;
+        } catch (final RuntimeException writeException) {
+            if (newTableCreated) {
+                // Delete it to avoid leaving a partial table in the catalog
+                try {
+                    catalog.dropTable(tableIdentifier, true);
+                } catch (final RuntimeException dropException) {
+                    writeException.addSuppressed(dropException);
+                }
+            }
+            if (newNamespaceCreated) {
+                // Delete it to avoid leaving a partial namespace in the catalog
+                dropNamespaceIfExists(tableIdentifier.namespace());
+            }
+            throw writeException;
+        }
+    }
+
+    private static IcebergParquetWriteInstructions verifyWriteInstructions(
+            @Nullable final IcebergWriteInstructions instructions) {
+        if (instructions == null) {
+            return IcebergParquetWriteInstructions.DEFAULT;
+        }
+        // We ony support writing to Parquet files
+        if (!(instructions instanceof IcebergParquetWriteInstructions)) {
+            throw new IllegalArgumentException("Unsupported instructions of class " + instructions.getClass() + " for" +
+                    " writing Iceberg table, expected: " + IcebergParquetWriteInstructions.class);
+        }
+        return (IcebergParquetWriteInstructions) instructions;
+    }
+
+    private boolean createNamespaceIfNotExists(@NotNull final Namespace namespace) {
+        if (catalog instanceof SupportsNamespaces) {
+            final SupportsNamespaces nsCatalog = (SupportsNamespaces) catalog;
+            if (!nsCatalog.namespaceExists(namespace)) {
+                nsCatalog.createNamespace(namespace);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private void dropNamespaceIfExists(@NotNull final Namespace namespace) {
+        if (catalog instanceof SupportsNamespaces) {
+            final SupportsNamespaces nsCatalog = (SupportsNamespaces) catalog;
+            nsCatalog.dropNamespace(namespace);
+        }
+    }
+
+    private org.apache.iceberg.Table createNewIcebergTable(
+            @NotNull final TableIdentifier tableIdentifier,
+            @NotNull final SpecAndSchema specAndSchema,
+            @NotNull final IcebergParquetWriteInstructions writeInstructions) {
+        if (catalog.tableExists(tableIdentifier)) {
+            throw new IllegalArgumentException("Table already exists: " + tableIdentifier);
+        }
+        return catalog.createTable(tableIdentifier, specAndSchema.schema(), specAndSchema.partitionSpec(), null,
+                Map.of(TableProperties.DEFAULT_FILE_FORMAT, DEFAULT_GENERATED_FILE_FORMAT,
+                        TableProperties.PARQUET_COMPRESSION,
```

Comment:
Is it actually the case that all Parquet files in a Table must have the same compression codec?